src/DynamoCore/Graph/Notes/NoteModel.cs

```diff
@@ -40,11 +48,13 @@ public NodeModel PinnedNode
             get { return pinnedNode; }
             set
             {
+                CreatedPinNode = true;
```

Comment:
instead of creating a new member, can't you just check that this is not null?

Reply:
In this case, checking if `pinnedNode` is null will not work because I need to check if It's the first action after pinning the note.

Reply:
@filipeotero Is it because without check, we could also accidentally undo the text input of note? Would you eleborate?